lib/fluent/plugin/out_papertrail.rb

```diff
@@ -1,30 +1,84 @@
-require 'syslog_protocol'
-require 'fluent/output'
-
 module Fluent
-  class Papertrail < Fluent::Output
-    attr_accessor :socket, :papertrail_host, :papertrail_port
+  class Papertrail < Fluent::BufferedOutput
+    attr_accessor :socket
```

Comment:
Might as well make this `attr_reader` unless there's a good reason to create a setter.

Reply:
I had this going both ways because in the unit tests I'm overriding socket with a "TestSocket" and then reading from it. Is there a better way to override this on the class only during testing?

Reply:
If I were writing this, I'd have the socket entity implemented as a singleton via a class instance variable on a class called something like `LogMessage`. I'd then have a class method on that class that allowed me to set the socket to what I want given a context, and a default code path that reduced the need for manual setup (i.e. by reading from a config or something).

```ruby
class LogMessage
  @socket = default_socket

  def self.default_socket
    #setup @socket from config
  end

  def self.set_socket(tcp_socket)
    # @socket = tcp_socket
  end

  #
  # Other message/connection concerns
  #
end
```